core/src/main/java/org/elasticsearch/action/index/IndexResponse.java

```diff
@@ -36,42 +36,43 @@
  */
 public class IndexResponse extends DocWriteResponse {
 
-    private boolean created;
-
     public IndexResponse() {
 
     }
 
+    @Override
+    public Operation getOperation() {
+        return isCreated() ? Operation.CREATE : Operation.INDEX;
+    }
+
     public IndexResponse(ShardId shardId, String type, String id, long version, boolean created) {
-        super(shardId, type, id, version);
-        this.created = created;
+        super(shardId, type, id, version, toOperation(created));
+    }
+
+    public static Operation toOperation(boolean created) {
+        return created ? Operation.CREATE : Operation.INDEX;
     }
 
     /**
      * Returns true if the document was created, false if updated.
      */
     public boolean isCreated() {
-        return this.created;
+        return this.operation == Operation.CREATE;
     }
 
     @Override
     public RestStatus status() {
-        if (created) {
-            return RestStatus.CREATED;
-        }
-        return super.status();
+        return isCreated() ? RestStatus.CREATED : super.status();
```

Comment:
This is different for IndexResponse/UpdateResponse and DeleteResponse; index and update check for Operation.CREATED, whereas DeleteResponse is looking for the absence of Operation.DELETED. We could make this one default and override it in DeleteResponse though, although it might be more explicit to have them in each subclass?